crates/cli/src/main.rs

```diff
@@ -317,10 +317,43 @@ async fn main() -> anyhow::Result<()> {
     }
     let mut server_handles = Vec::with_capacity(config.host.len());
     for host in &config.host {
-        let addr = SocketAddr::new(*host, config.port);
-        let server = server_builder.clone().build(addr).await;
-        config.port = server.local_addr().port();
-        server_handles.push(server.run());
+        let mut addr = SocketAddr::new(*host, config.port);
+
+        match server_builder.clone().build(addr).await {
+            Ok(server) => {
+                config.port = server.local_addr().port();
+                server_handles.push(server.run());
+            }
+            Err(err) => {
+                tracing::info!(
+                    "Failed to bind to address {}:{}: {}. Retrying with a different port...",
+                    host,
+                    config.port,
+                    err
+                );
```

Comment:
Hmm, I don't think this is a good fallback. If I were running a tool in CI and saw this behaviour I would have been unpleasantly surprised.

Reply:
Hm, indeed. I though for a second that `anvil` does that -- but no:

```
anvil
Error: Address already in use (os error 98)

Location:
    /home/runner/work/foundry/foundry/crates/anvil/src/lib.rs:229:28
```

I guess we have to replace the fallback with just error propagation.